src/test/regress/sql/resync_metadata.sql

```diff
@@ -0,0 +1,132 @@
+CREATE SCHEMA resync_metadata;
+SET search_path TO resync_metadata;
+
+CREATE TABLE test_serial(a bigserial PRIMARY KEY);
+SELECT create_distributed_table('test_serial', 'a');
+
+CREATE SEQUENCE myseq;
+CREATE TABLE test_sequence(a bigint DEFAULT nextval('myseq'));
+SELECT create_distributed_table('test_sequence', 'a');
+
+CREATE SEQUENCE myseq_ref;
+CREATE TABLE test_serial_ref(a bigserial PRIMARY KEY, b bigint DEFAULT nextval('myseq_ref'));
+SELECT create_reference_table('test_serial_ref');
+
+INSERT INTO test_serial_ref VALUES(DEFAULT) RETURNING *;
+INSERT INTO test_serial_ref VALUES(DEFAULT) RETURNING *;
+
+SET client_min_messages TO ERROR;
+SELECT 1 FROM citus_add_node('localhost', :master_port, groupid:=0);
+
+CREATE SEQUENCE myseq_locl_to_dist;
+CREATE TABLE test_local_to_dist(a bigserial PRIMARY KEY, b bigint DEFAULT nextval('myseq_locl_to_dist'));
+SELECT citus_add_local_table_to_metadata('test_local_to_dist');
+INSERT INTO test_local_to_dist VALUES(DEFAULT) RETURNING *;
+INSERT INTO test_local_to_dist VALUES(DEFAULT) RETURNING *;
+
+SET citus.shard_replication_factor TO 1;
+CREATE SEQUENCE other_id_seq;
+CREATE TABLE sensors(
+measureid bigserial,
+other_id bigint DEFAULT nextval('other_id_seq'),
+eventdatetime date) PARTITION BY RANGE(eventdatetime);
+
+CREATE TABLE sensors_old PARTITION OF sensors FOR VALUES FROM ('2000-01-01') TO ('2020-01-01');
+CREATE TABLE sensors_2020_01_01 PARTITION OF sensors FOR VALUES FROM ('2020-01-01') TO ('2020-02-01');
+CREATE TABLE sensors_news PARTITION OF sensors FOR VALUES FROM ('2020-05-01') TO ('2025-01-01');
+SELECT create_distributed_table('sensors', 'measureid', colocate_with:='none');
+
+\c - - - :worker_1_port
+SET search_path tO resync_metadata;
+
+INSERT INTO test_serial VALUES(DEFAULT) RETURNING *;
+INSERT INTO test_serial VALUES(DEFAULT) RETURNING *;
+
+INSERT into test_sequence VALUES(DEFAULT) RETURNING *;
+INSERT into test_sequence VALUES(DEFAULT) RETURNING *;
+
+INSERT INTO test_local_to_dist VALUES(DEFAULT) RETURNING *;
+INSERT INTO test_local_to_dist VALUES(DEFAULT) RETURNING *;
+
+INSERT INTO test_serial_ref VALUES(DEFAULT) RETURNING *;
+INSERT INTO test_serial_ref VALUES(DEFAULT) RETURNING *;
+
+INSERT INTO sensors VALUES (DEFAULT, DEFAULT, '2010-01-01') RETURNING *;
+INSERT INTO sensors_news VALUES (DEFAULT, DEFAULT, '2021-01-01') RETURNING *;
+
+\c - - - :master_port
+SELECT start_metadata_sync_to_node('localhost', :worker_1_port);
+
+
+\c - - - :worker_1_port
+SET search_path tO resync_metadata;
+
+-- can continue inserting with the existing sequence/serial
+INSERT INTO test_serial VALUES(DEFAULT) RETURNING *;
```

Comment:
In addition to insert tests, can we also check nextval just to show that doesn't change explicitly?

Reply:
I don't think that'd add much value here -- besides being a little confusing as PG already does `nextval` under the hood for both bigserial and regular sequences?
```SQL
CREATE TABLE test_serial_ref(a bigserial PRIMARY KEY, b bigint DEFAULT nextval('myseq_ref'));

\d test_serial_ref
                           Table "public.test_serial_ref"
 Column |  Type  | Collation | Nullable |                  Default                   
--------+--------+-----------+----------+--------------------------------------------
 a      | bigint |           | not null | nextval('test_serial_ref_a_seq'::regclass)
 b      | bigint |           |          | nextval('myseq_ref'::regclass)

```